package/src/components/Message/Message.tsx

```diff
@@ -52,21 +52,36 @@ import {
 } from '../MessageList/hooks/useMessageList';
 import type { MessageActionListItemProps } from '../MessageOverlay/MessageActionListItem';
 
+export type Emitter =
+  | 'card'
+  | 'fileAttachment'
+  | 'gallery'
+  | 'giphy'
+  | 'message'
+  | 'messageContent'
+  | 'messageReplies'
+  | 'reactionList'
+  | 'textLink'
+  | 'textMention';
```

Comment:
you don't need `textMention` here!!

Reply:
When I remove this and try using it in the sample app after building the type I get an error that says the types of `Emitter | undefined` and `textMention` don't overlap. I think it's needed or maybe there's something I am missing @vishalnarkhede

Reply:
I think`textMention` should be part of the union for this to work properly

Reply:
just change TouchableHandlerPayload to following

```tsx
export type TouchableHandlerPayload = {
  defaultHandler?: () => void;
  event?: GestureResponderEvent;
} & (
  | {
      additionalInfo?: Record<string, unknown>;
      emitter?: Emitter;
    }
  | TextMentionTouchableHandlerPayload
);
```

Reply:
✅